packages/ember-glimmer/tests/integration/binding_integration_test.js

```diff
@@ -0,0 +1,72 @@
+import { RenderingTest, moduleFor } from '../utils/test-case';
+import { Component } from '../utils/helpers';
+import { set } from 'ember-metal/property_set';
+import { Binding } from 'ember-metal/binding';
+
+moduleFor('Binding integration tests', class extends RenderingTest {
+
+  ['@test bound values should update after the parent element is removed and then re-appended']() {
+    let instance;
+    let FooBarComponent = Component.extend({
+      init() {
+        this._super(...arguments);
+        instance = this;
+      }
+    });
+
+    this.registerComponent('foo-bar', { ComponentClass: FooBarComponent, template: '{{#if showStuff}}{{boundValue}}{{else}}Not true.{{/if}}' });
+    this.render('{{foo-bar showStuff=showStuff boundValue=boundValue}}', {
+      showStuff: true,
+      boundValue: 'foo'
+    });
+    this.assertText('foo');
+    this.assertStableRerender();
+
+    this.runTask(() => set(this.context, 'showStuff', false));
+    this.assertText('Not true.');
+    this.runTask(() => set(this.context, 'showStuff', true));
+    this.assertText('foo');
+
+    this.runTask(() => {
+      instance.remove();
+      set(this.context, 'showStuff', false)
+    });
+    this.runTask(() => {
+      set(this.context, 'showStuff', true);
+      this.rerender();
+    });
+
+    this.runTask(() => set(this.context, 'boundValue', 'bar'));
+    this.assertText('bar');
```

Comment:
this is a migrated test and it won't even work in HTMLBars. The old test was using a controller, which I'm not using here, and since `this.context` proxies `this.component` (which was the parentView in the old test), I can't really call `this.component.remove()` otherwise I won't be able to set `this.context` after removing it.